android/src/main/java/com/oney/WebRTCModule/WebRTCModule.java

```diff
@@ -485,6 +497,281 @@ MediaConstraints constraintsForOptions(ReadableMap options) {
+                    return null;
+                }
+
+                RtpTransceiver transceiver = null;
+                if (options.hasKey("type")) {
+                    String kind = options.getString("type");
+                    transceiver = pco.addTransceiver(mediaTypeForString(kind), parseTransceiverOptions(options.getMap("init")));
+                } else if (options.hasKey("trackId")) {
+                    String trackId = options.getString("trackId");
+                    MediaStreamTrack track = getTrack(trackId);
+                    transceiver = pco.addTransceiver(track, parseTransceiverOptions(options.getMap("init")));
+
+                } else {
+                    // This should technically never happen as the JS side checks for that.
+                    Log.d(TAG, "peerConnectionAddTransceiver() no type nor trackId provided in options");
+                    return null;
+                }
+
+                // We need the timestamp to reorder the transceivers array 
+                Long tsLong = System.currentTimeMillis()/1000;
+                String ts = tsLong.toString();      
+                if (transceiver == null) {
+                    Log.d(TAG, "peerConnectionAddTransceiver() Error adding transceiver");
+                    return null;
+                }
+                WritableMap params = Arguments.createMap();
+                params.putString("timestamp", ts);
+                params.putMap("transceiver", pco.serializeTransceiver(transceiver));
+                return params;
+            }).get();
+        } catch (InterruptedException | ExecutionException e) {
+            Log.d(TAG, "peerConnectionAddTransceiver() " + e.getMessage());
+            return null;
+        }
+    }
+
+    @ReactMethod(isBlockingSynchronousMethod = true)
+    public WritableMap peerConnectionAddTrack(int id,
+                                              String trackId,
+                                             ReadableMap options) {
+        try {
+            return (WritableMap) ThreadUtils.submitToExecutor((Callable<Object>) () -> {
+                PeerConnectionObserver pco = mPeerConnectionObservers.get(id);
+                if (pco == null) {
+                    Log.d(TAG, "peerConnectionAddTrack() peerConnection is null");
+                    return null;
+                }
+
+                MediaStreamTrack track = getTrack(trackId);
+                List<String> streamIds = new ArrayList<>();
+                if (options.hasKey("streamIds")) {
+                    ReadableArray rawStreamIds = options.getArray("streamIds");
+                    if (rawStreamIds != null) {
+                        for (int i = 0; i < rawStreamIds.size(); i++) {
+                            streamIds.add(rawStreamIds.getString(i));
+                        }
+                    }
+                }
+                RtpSender sender = pco.getPeerConnection().addTrack(track, streamIds);
```

Comment:
We have a little problem which might require patching WebRTC. The returned `RtpSender` owns the track and will dispose it when the sender itself goes away.

I don't see how to change this. There is a `setTrack` API in the sender which allows to replace the track, and there is a `takeOwnership` parameter, but that will still dispose the previous track...

Reply:
Maybe not. Those objects are refcounted so it's just our instance of the track that will be gone. We need a fresh native object that represents the same entity. Not sure how to do that (yet) though.

Reply:
I am not quite sure this is the case. This function snippet from `RtpSender.java` seems to suggest the track is only disposed when it is owned by the sender (i.e. when setTrack has been used). 
```
  public void dispose() {
    checkRtpSenderExists();
    if (dtmfSender != null) {
      dtmfSender.dispose();
    }
    if (cachedTrack != null && ownsTrack) {
      cachedTrack.dispose();
    }
    JniCommon.nativeReleaseRef(nativeRtpSender);
    nativeRtpSender = 0;
  }
```

Reply:
But look at the constructor and the default value for `ownsTrack`:

```
/** Java wrapper for a C++ RtpSenderInterface. */
public class RtpSender {
  private long nativeRtpSender;

  @Nullable private MediaStreamTrack cachedTrack;
  private boolean ownsTrack = true;
  private final @Nullable DtmfSender dtmfSender;

  @CalledByNative
  public RtpSender(long nativeRtpSender) {
    this.nativeRtpSender = nativeRtpSender;
    long nativeTrack = nativeGetTrack(nativeRtpSender);
    cachedTrack = MediaStreamTrack.createMediaStreamTrack(nativeTrack);

    long nativeDtmfSender = nativeGetDtmfSender(nativeRtpSender);
    dtmfSender = (nativeDtmfSender != 0) ? new DtmfSender(nativeDtmfSender) : null;
  }
```

Reply:
So the following test: create a local video track, display it in a view. Create a PC and add it. Destroy the PC. Try to render the track again. Does it work?

Reply:
I did the following test on the example app `GumTestApp` as follows: I created a replace button which creates a new peer connection everytime and attaches the same track everytime. It was done on an Android Emulator. The app did not crash or cause any issues:
```js
  const [pc, setPc] = useState(new RTCPeerConnection());
  const replace = () => {
    console.log('replace');
    pc.close();
    setPc(new RTCPeerConnection());
    const track = stream.getTracks()[0];
    pc.addTrack(track, [stream]);
  }

```